test/e2e/kubelet.go

```diff
@@ -128,59 +131,207 @@ func updateNodeLabels(c clientset.Interface, nodeNames sets.String, toAdd, toRem
 	}
 }
 
+// Calls startVolumeServer to create and run a nfs-server pod. Returns server pod and its
+// ip address.
+// Note: startVolumeServer() waits for the nfs-server pod to be Running and sleeps some
+//   so that the nfs server can start up.
+func createNfsServerPod(c clientset.Interface, config VolumeTestConfig) (*v1.Pod, string) {
+
+	pod := startVolumeServer(c, config)
+	Expect(pod).NotTo(BeNil())
+	ip := pod.Status.PodIP
+	Expect(len(ip)).NotTo(BeZero())
+	framework.Logf("NFS server IP address: %v", ip)
+
+	return pod, ip
+}
+
+// Creates a pod that mounts an nfs volume that is served by the nfs-server pod. The container
+// will execute the passed in shell cmd. Waits for the pod to start. Returns the pod, its external
+// IP, and its UID.
+// Note: the nfs plugin is defined inline, no PV or PVC.
+func createPodUsingNfs(f *framework.Framework, c clientset.Interface, ns, nfsIP, cmd string) (*v1.Pod, string, string) {
+
+	By("create pod using nfs volume")
+
+	isPrivileged := true
+	cmdLine := []string{"-c", cmd}
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: registered.GroupOrDie(v1.GroupName).GroupVersion.String(),
+		},
+		ObjectMeta: v1.ObjectMeta{
+			GenerateName: "pod-nfs-vol-",
+			Namespace:    ns,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    "pod-nfs-vol",
+					Image:   "gcr.io/google_containers/busybox:1.24",
```

Comment:
Agree. I think maybe a new PR should be created with the sole purpose of creating these constants for the nfs-server pod, busybox, and storage containers such as glusterfs and ceph.